tests/test_db_manager.py

```diff
@@ -0,0 +1,165 @@
+from unittest import TestCase
+import pandas as pd
+# import psycopg2
+from cbcdb.main import DBManager
+
+
+class TestDBManager(TestCase):
+
+    @staticmethod
+    def _get_db_inst():
+        db = DBManager(
+            debug_output_mode=True,
+            use_ssh=False,
+            ssh_remote_bind_port=5434,
+            db_name='test',
+            db_user='test',
+            db_password='test',
+            db_schema='public',
+            db_host='localhost')
+        return db
+
+    def _make_some_data(self, db):
+        # Check if the table exists
+        test_table_name = 'color'
+
+        sql = f"""
+        select exists(
+               SELECT
+               FROM pg_catalog.pg_class c
+                        JOIN pg_catalog.pg_namespace n ON n.oid = c.relnamespace
+               WHERE n.nspname = 'public'
+                 AND c.relname = '{test_table_name}'
+           );"""
+
+        res = db.get_single_result(sql)
+
+        if res[0]:
+            # Delete the database
+            sql = f"drop table public.{test_table_name}"
+            db.execute_simple(sql)
+
+        # sql = """
+        # create table public.testing
+        #        (id             serial,
+        #         test           varchar(10),
+        #         a_number       smallint,
+        #         firstname      varchar(75),
+        #         lastname       varchar(75),
+        #         primary key (id));"""
+
+        sql = f"""CREATE TABLE public.{test_table_name} (
+                    color_id serial,
+                    color_name VARCHAR NOT NULL
+                );"""
+
+        db.execute_simple(sql)
+        sql = f"""
+        insert into public.{test_table_name}(color_name)
+        values ('red'),('green'),('pink'),('purple'),('blue'),('cyan')
+        """
+        db.execute_simple(sql)
+
+        return test_table_name
+
+    # def test__get_connection(self):
+    #     db = self._get_db_inst()
+    #     conn = db._get_connection()
+    #     self.assertEqual(0, conn.closed)
+    #     # db._safe_tunnel_close(conn)
+
+    def test__get_sql_dataframe(self):
+        db = self._get_db_inst()
+        table_name = self._make_some_data(db)
+
+        res = db.get_sql_dataframe(f'select * from {table_name}')
+        golden = 'red'
```

Comment:
Not sure I understand this variable name, but the test looks good!

Reply:
Golden is a term I picked up in the semiconductor industry. We would create golden sample (chips we had fully tested and characterized). The goldens would be used to compare to other chips under test. If there's a more industry standard variable name I would like to switch to that.

Reply:
I see! I don't think there's an industry standard variable name here. In these cases, I would actually just put the desired string literal directly into the assertion (i.e. `assertEqual(res, 'red')`), since it's short and I don't think any clarity is added by having an intermediate variable.

I'd probably only use an intermediate variable if the desired result is very long. And in that case I'd recommend using a very clear name like `desired_result` or something.